CT-1687 filter open cases by timeliness

misaka opened the pull request.

aliuk2012 commented on app/models/assignment.rb:
```diff
@@ -10,6 +10,7 @@
 #  updated_at :datetime         not null
 #  role       :enum
 #  user_id    :integer
+#  approved   :boolean          default(FALSE)
```
Does this need to be here?

aliuk2012 commented on db/structure.sql:
```diff
@@ -129,7 +129,8 @@ CREATE TABLE assignments (
     role team_roles,
-    user_id integer
+    user_id integer,
+    approved boolean DEFAULT false
```
Why are we adding "approved" to a unique team validation?

aliuk2012 commented on db/structure.sql:
```diff
@@ -943,6 +944,7 @@ INSERT INTO "schema_migrations" (version) VALUES
 ('20170420122223'),
-('20170424133127');
+('20170424133127'),
+('20170523131602');
```
If we dont need approved then this should go too

aliuk2012 commented on spec/factories/assignments.rb:
```diff
@@ -10,6 +10,7 @@
 #  updated_at :datetime         not null
 #  role       :enum
 #  user_id    :integer
+#  approved   :boolean          default(FALSE)
```
Does this need to be here?

aliuk2012 commented on spec/models/assignment_spec.rb:
```diff
@@ -10,6 +10,7 @@
 #  updated_at :datetime         not null
 #  role       :enum
 #  user_id    :integer
+#  approved   :boolean          default(FALSE)
```
Does this need to be here?

misaka committed:
CT-1687 allow controller to specify cases when filtering lists

misaka committed:
CT-1687 remove in_time and late tabs from all open cases

This filtering will be handled by the new-style filters being added as part of
this same ticket.

misaka committed:
CT-1687 add timeliness filter to open cases list

misaka committed:
CT-1687 update other filter tests for new timeliness filter

misaka committed:
CT-1687 remove list_params field from SearchQuery

This was being use do reconstitute the GlobalNavManager, but we've changed the
system so that it passes an instance of the GlobalNavManagar. list_params may
not even have been necessary since the necessary params can be derived from the
list_path.

misaka committed:
CT-1687 reorder timeliness in filter tabs

misaka committed:
CT-1687 make assigned_to filter submit to current URL

misaka committed:
CT-1687 remove unused params and allow StandardSetup to be biiig

Rubocop was complaining ...
